web/src/shared/components/ourdna/BarChart.tsx

```diff
@@ -0,0 +1,251 @@
+            .domain([0, d3.max(transformedData, (d) => d.count) || 1])
+            .nice()
+            .rangeRound([height, 0])
+
+        const color = d3
+            .scaleOrdinal<string>()
+            .domain(transformedData.map((d) => d.site))
+            .range(Object.values(ourdnaColours))
+
+        groupedData.forEach(([hour, values]) => {
+            if (x0(hour) === undefined) {
+                return
+            }
+
+            const x1 = d3
+                .scaleBand()
+                .domain(values.map((d) => d.site))
+                .rangeRound([0, x0.bandwidth()])
+                .padding(0.05)
+
+            if (Number.isNaN(x0(hour))) {
+                return
+            }
+
+            g.append('g')
+                .attr('transform', `translate(${x0(hour)},0)`)
+                .selectAll('rect')
+                .data(values)
+                .enter()
+                .append('rect')
+                .attr('x', (d) => {
+                    const x = x1(d.site)
+                    if (Number.isNaN(x)) {
+                        return 0
+                    }
+                    return x
+                })
+                .attr('y', (d) => y(d.count))
+                .attr('width', x1.bandwidth())
+                .attr('height', (d) => Math.max(0, height - y(d.count)))
+                .attr('fill', (d) => color(d.site))
+        })
+
+        g.append('g')
+            .attr('class', 'axis-label')
+            .attr('transform', `translate(0,${height})`)
+            .call(d3.axisBottom(x0))
+            .selectAll('text')
+            .attr('dy', '1em') // Adjust the position of the x-axis labels
+
+        g.append('g')
+            .attr('class', 'axis-label')
+            .call(d3.axisLeft(y).ticks(y.domain()[1]).tickFormat(d3.format('d')))
+            .selectAll('text')
+            .attr('dx', '-1em') // Adjust the position of the y-axis labels
+
+        g.append('text')
+            .attr('class', 'axis-label')
+            .attr('text-anchor', 'end')
+            .attr('x', width / 2)
+            .attr('y', height + margin.bottom - 15)
+            .text('Hours Taken')
+
+        g.append('text')
+            .attr('class', 'axis-label')
+            .attr('text-anchor', 'end')
+            .attr('transform', 'rotate(-90)')
+            .attr('x', -height / 2)
+            .attr('y', -margin.left + 15)
+            .text('Sample Count')
+
+        const legend = svg
+            .append('g')
+            .attr('font-family', 'sans-serif')
+            .attr('font-size', 10)
+            .attr('text-anchor', 'start')
+            .attr('transform', `translate(${dimensions.width - margin.right + 10},${margin.top})`)
+            .selectAll('g')
+            .data(transformedData.map((d) => d.site).filter((v, i, a) => a.indexOf(v) === i))
+            .enter()
+            .append('g')
+            .attr('transform', (d, i) => `translate(0,${i * 20})`)
+
+        // Append X axis
```

Comment:
Not sure if you saw but d3 has a helper lib for building axes, https://d3js.org/d3-axis. This is fine for now though

Reply:
lol, sorry, just realised you're already doing this, ignore me